azerothcore/azerothcore-wotlk
# fix(Scripts/Spells): Finger of Frost Shatter Combo.

Fixes #11338
Fixes #11876



## Issues Addressed:

- Closes #11338
- Closes #11876
- Closes https://github.com/chromiecraft/chromiecraft/issues/3123
- Closes https://github.com/chromiecraft/chromiecraft/issues/3542

## Tests Performed:

- Not tested.

## How to Test the Changes:

Combo Shatter
1. Go to a dummy with only Fingers of Frost specced,
2. When at only 1 charge of Fingers of Frost, attempt (and fail) to cast Deep Freeze.

Cast on hit proc
1. `.aura 44545`
2. cast Frostbolt (ID: 116) or Frostfire Bolt (ID: 44614) until you see a proc on-cast and a proc on-hit


## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.
